pandas/io/formats/format.py

```diff
@@ -1533,7 +1533,14 @@ def format_values_with(float_format):
     def _format_strings(self) -> List[str]:
         # shortcut
         if self.formatter is not None:
```

Comment:
actually this entire section you added should be instead on L1440

Reply:
and you will want to do something like
```
mask = isna(values)	
values = np.array(values, dtype="object")
values[mask] = na_rep
imask = (~mask).ravel(
values.flat[imask] = np.array(	
                [formatter(val) for val in values.ravel()[imask]]
            )
```

you will want to factor that out into a function and use it above in 2 (or more places)

Reply:
I'm not sure I understand, `get_result_as_array` never actually gets called here. Are you thinking the formatter should already be handling NaN?

Reply:
no my point is to share code; you are doing virtually the same thing, just in another way

Reply:
Ok I see what you mean